src/System.IO.Ports/tests/Support/PortHelper.cs

```diff
@@ -68,7 +69,7 @@ public static string[] GetPorts()
             {
                 foreach (string str in retval)
                 {
-                    if (str.StartsWith("COM"))
+                    if (str.Length > 3 && str.Length < 7 && str.StartsWith("COM") && str.Substring(3).All(c => char.IsDigit(c)))
```

Comment:
Would a regex would better capture the intent here?  Something like `COM\d+$` perhaps?